### PR TITLE
Add preview annotation and update .gitignore

### PR DESCRIPTION
**Pull request by ilanbiala**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,8 +3,7 @@
 .idea/
 out/
 target/
-java/
```
Comment (efossier) [inline .gitignore]: why did you remove this? java typically contains javadocs which I dont think we want to check in.

Comment (ilanbiala) [inline .gitignore]: @jingming pointed out that it also ignores source files in `src/main/java/`, which isn't ideal. Where do the docs show up in `java` folder?

File: src/main/java/com/twilio/annotations/Beta.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.CLASS)
+@Target({
+  ElementType.ANNOTATION_TYPE,
+  ElementType.CONSTRUCTOR,
+  ElementType.FIELD,
+  ElementType.METHOD,
+  ElementType.TYPE
```
Comment (efossier) [inline src/main/java/com/twilio/annotations/Beta.java]: might want to add ElementType.PACKAGE

File: .gitignore (hunk 1 above)
Comment (efossier) [inline .gitignore]: generated javadoc files typically show up in `java/{version}/`. Will making this `./java` fix it?

**Comment by ilanbiala** [comment]
Annotations are unnecessary, gitignore in a separate commit.
